staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -766,6 +806,45 @@ func ValidateCustomResourceColumnDefinition(col *apiextensions.CustomResourceCol
 	return allErrs
 }
 
+func ValidateCustomResourceSelectableFields(schema *structuralschema.Structural, uniqueSelectableFields sets.Set[string], selectableField *apiextensions.SelectableField, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if len(selectableField.JSONPath) == 0 {
```

Comment:
following the logic, the nesting reduces to 

```
if len() == 0{
  return because there's nothing else to check
}

path, foundSchema, err := validate
if err !=nil{
  return because there is nothing else to check
} 
if path
if allowable
if unique

return all errs

```

with early returns recommended by go?